apiserver/common/machine.go

```diff
@@ -161,12 +162,20 @@ func ModelMachineInfo(st ModelManagerBackend) (machineInfo []params.ModelMachine
 				mInfo.HAPrimary = &isPrimary
 			}
 		}
-		instId, displayName, err := m.InstanceNames()
+		machineUUID, err := machineService.GetMachineUUID(ctx, machine.Name(m.Id()))
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		displayName, err := machineService.InstanceName(ctx, machineUUID)
+		if err != nil && !errors.Is(err, machineerrors.NotProvisioned) {
+			return nil, errors.Trace(err)
+		}
+		instId, err := machineService.InstanceID(ctx, machineUUID)
```

Comment:
We should have a method to get the InstanceName and the ID at the same time. This is very potato.

Reply:
That's what we had in the legacy state, but I found it suboptimal: the display name is *only* used in the full status, whilst the `InstanceID()` method is used everywhere. This would add a bigger struct to de serialize with sqlair and it would also mean that we'll have `instanceID, _, err := instanceNames()` everywhere